mono/metadata/file-mmap-posix.c

```diff
@@ -275,10 +277,12 @@ open_file_map (const char *c_path, int input_fd, int mode, gint64 *capacity, int
 		}
 	}
 
+	MONO_ENTER_GC_SAFE;
 	if (c_path) //FIXME use io portability?
 		fd = open (c_path, file_mode_to_unix (mode) | access_mode_to_unix (access), DEFAULT_FILEMODE);
 	else
 		fd = dup (input_fd);
```

Comment:
`dup`?

Reply:
I thiiiiiiink `dup` calls `fcntl` internally? I should probably verify one way or another.